### ISSUE TITLE
IIFE with try/catch statement inside causes unreachable code

### ISSUE THREAD

**Issue by toiglak** [microsoft/TypeScript#38206]
**TypeScript Version:**  3.8.3

**Search Terms:**
false positive iife immediately invoked function expression try catch block unreachable code detected 7027

**Expected behavior:**
Not throwing `Unreachable code detected. (7027)` error.

**Actual behavior:**
Throwing false-positive `Unreachable code detected. (7027)` error.

**Related Issues:** 
I asked about it on [stackoverflow](https://stackoverflow.com/questions/61058410/false-positive-unreachable-code-detected-ts7027) and I didn't find any similar issues on this repo.

**Code**
```js
const foo = (() => {
  try {
    return { val: 4 };
  } catch (err) {
    return { val: -1}
  }
})();

console.log(foo.val); // unreachable code
```

<details><summary><b>Output</b></summary>

```ts
"use strict";
const foo = (() => {
    try {
        return { val: 4 };
    }
    catch (err) {
        return { val: -1 };
    }
})();
console.log(foo.val);

```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "useDefineForClassFields": false,
    "alwaysStrict": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "downlevelIteration": false,
    "noEmitHelpers": false,
    "noLib": false,
    "noStrictGenericChecks": false,
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "esModuleInterop": true,
    "preserveConstEnums": false,
    "removeComments": false,
    "skipLibCheck": false,
    "checkJs": false,
    "allowJs": false,
    "declaration": true,
    "experimentalDecorators": false,
    "emitDecoratorMetadata": false,
    "target": "ES2017",
    "module": "ESNext"
  }
}
```

</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/play#code/MYewdgzgLgBAZiEMC8MAUaCUKB8MDeAUDDFAE4CeBxJMZAplAK5lgEwBuAhgDYBcMACwwAvgG4aImMC5RgAC3T0yZbEVp1GLNvk68BAWgCMIyYRGYsEwqEgge9AHQ8QAczQIQj7j0xigA)

**Comment by IllusionMH** [comment]
Duplicate of #36828

Already fixed in 3.9 (check Nightly in Playground)

**Comment by toiglak** [comment]
Thanks!
